src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -522,13 +553,17 @@ private function addServiceInclude(string $cId, Definition $definition): string
             }
 
             foreach (array_diff_key(array_flip($lineage), $this->inlinedRequires) as $file => $class) {
+                $file = preg_replace('#^\\$this->targetDirs\[(\d++)\]#', sprintf('\dirname(__DIR__, %d + $1)', $this->asFiles), $file);
```

Comment:
why generating code using `$this->targetDirs` first, to replace it later with `dirname(__DIR__)` ? We would generate the right code directly.

Reply:
I prefer not changing this when it's not needed: the code starts with `$this->targetDirs[0] = \dirname($containerDir)`: targets dir are dynamic. This is used when calling cache clear and is stable, no reason to change.
Using dirname() is really useful when referencing files in the src/ or in the vendor/ folder, which is where it is used.
Lowest risk strategy :)

Reply:
Second look, I did something, please check #33529 :)